version 1

tsohleDev opened the pull request:
# Recipe recipe 

In this PR we:
- Create a database from an ERD
- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipe list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

gitbygaurav committed:
Created new rails app

gitbygaurav committed:
Add rspec devise and cancancan gem

gitbygaurav committed:
Add rubocop and fix linter errors

gitbygaurav committed:
Add stylelint

gitbygaurav committed:
Merge pull request #12 from gauravgangwarweb/initial-setup

Recipe-app: Initial setup (Group Task)

gitbygaurav committed:
Add devise for authentication

gitbygaurav committed:
Add styling for registration page and login page

gitbygaurav committed:
Fix styellint errors

gitbygaurav committed:
Add suggested changes

gitbygaurav committed:
Merge pull request #13 from gauravgangwarweb/authentication

Recipe-app: Login page and registration page

gitbygaurav committed:
Add recipe controller

gitbygaurav committed:
Add recipe model

gitbygaurav committed:
Add views for recipe

gitbygaurav committed:
Fix linter errors

gitbygaurav committed:
Merge pull request #14 from gauravgangwarweb/recipe

Recipe-app: Recipes list

gitbygaurav committed:
Add navbar markup

gitbygaurav committed:
Add styling for navbar

gitbygaurav committed:
Fix linter errors

tsohleDev committed:
scufle up Foods

tsohleDev committed:
merge

tsohleDev committed:
fix rubocop linters

gitbygaurav committed:
Merge pull request #15 from gauravgangwarweb/navigation-bar

Recipe-app: Create a navigation

tsohleDev committed:
Add reference in the model

tsohleDev committed:
Use current_user to create foods

tsohleDev committed:
Merge pull request #16 from gauravgangwarweb/food-list

scufle up Foods

tsohleDev committed:
Create public recipes

tsohleDev committed:
Include email confirmation again

tsohleDev committed:
Skip auth for public

tsohleDev committed:
Shooping-list

tsohleDev committed:
Shooping-list

gitbygaurav committed:
Merge pull request #17 from gauravgangwarweb/public-reciepes

Create public recipes

gitbygaurav committed:
Merge pull request #18 from gauravgangwarweb/general-shoping-cart

General shoping cart

gitbygaurav committed:
Add ability class

gitbygaurav committed:
Add recipe food controller and recipe details view

gitbygaurav committed:
Fix linter errors

tsohleDev committed:
Foods specs

gitbygaurav committed:
Merge pull request #19 from gauravgangwarweb/tests

Tests

gitbygaurav committed:
Add tests for recepie

gitbygaurav committed:
Merge pull request #20 from gauravgangwarweb/Tests-recipe

Add tests for recepie

gitbygaurav committed:
Add readme file

codepantha commented on app/controllers/foods_controller.rb:
```diff
@@ -0,0 +1,77 @@
+  # GET /foods/1 or /foods/1.json
+  def show; end
+
+  def shooping_list
+    @foods = Food.all.where(user: current_user)
+    render :shooping_list
+  end
```
- [ ] I noticed that your implementation of the shopping list has not been done in accordance with the project requirements. You are simply displaying all the foods in your database. Kindly see the attached below 👇 

![image](https://user-images.githubusercontent.com/46839548/234908739-7ac1065a-ee9f-458e-a13f-1d736d84ae17.png)

codepantha commented on db/migrate/20230425093411_create_foods.rb:
```diff
@@ -0,0 +1,13 @@
+      t.string :measurement_unit
+      t.decimal :price
+      t.integer :quantity
+      t.integer :user_id
+
+      t.timestamps
+    end
```
- [ ] The`user_id` is a foreign key therefore it is best practice to make sure it is indexed because indexing makes columns faster to query 👍

codepantha commented on spec/views/foods/index.html.erb_spec.rb:
```diff
@@ -0,0 +1,46 @@
+      expect(page).to have_content 'Add food'
+    end
+
+    it 'show food name' do
+      visit foods_path
+      expect(page).to have_content @food1.name
+    end
+
+    it 'show food measurement unit ' do
+      visit foods_path
+      expect(page).to have_content @food1.measurement_unit
+    end
```
- [ ] These 2 tests are failing ❌ Please try to make them pass ✔️ 
![image](https://user-images.githubusercontent.com/46839548/234914332-23478f53-7432-482e-94e1-3231619422ab.png)

codepantha commented on spec/views/recipes/index.html.erb_spec.rb:
```diff
@@ -0,0 +1,26 @@
+    expect(page).to have_content 'Recipe'
+  end
+
+  it 'click on new recipe' do
+    click_on 'Add new recipe'
+    expect(current_path).to eql recipes_path
+  end
```
- [ ] This test is currently failing because the expected path does not match the actual path. Please try to make it green ✔️ 
![image](https://user-images.githubusercontent.com/46839548/234915239-f38b2666-73a0-414d-9f52-fe505756704b.png)

codepantha reviewed (changes requested):
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media1.giphy.com/media/rZ10JtoPRv1Ly/giphy.gif"/>

## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tsohleDev committed:
Fixes

Fix shopping list to show user foods only
fix view of foods/index
Rewrite the test

elisha2kyakpo1 reviewed (approved):
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to make the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
